pkg/server/sessions/session.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2021 CodeNotary, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sessions
+
+import (
+	"context"
+	"github.com/codenotary/immudb/embedded/sql"
+	"github.com/codenotary/immudb/pkg/api/schema"
+	"github.com/codenotary/immudb/pkg/auth"
+	"github.com/codenotary/immudb/pkg/database"
+	"github.com/codenotary/immudb/pkg/logger"
+	"github.com/codenotary/immudb/pkg/server/sessions/internal/transactions"
+	"github.com/rs/xid"
+	"google.golang.org/grpc/metadata"
+	"sync"
+	"time"
+)
+
+type Status int64
+
+const (
+	ACTIVE Status = iota
+	IDLE
+	DEAD
+)
+
+type Session struct {
```

Comment:
session may have a lock, but it's not actually an extension of a rwlock, I'd suggest having it as an attribute instead

Reply:
didn't get this. Sessions has a RWLock

Reply:
the Session struct is extending RWLock while it could just have a private attribute for it. Otherwise session struct could be used somewhere else as a lock